app/services/github/contributions/pull_request_collector.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Github
+  module Contributions
+    class PullRequestCollector
+      ORG_INDEX = 3
+      REPO_INDEX = 4
+      PR_INDEX = 6
+
+      def initialize(client:)
+        @client = client
+      end
+
+      def all
+        valid_pull_requests = Contribution.where.not(state: :refused).select(:id, :link, :pr_state)
+        updated_pull_request = valid_pull_requests.map {|contribution| { id: contribution.id, pr_state: new_pr_state(contribution.link)} }
+
+        updated_pull_request.select{ |new_pr| validate_update(valid_pull_requests.find_by(id: new_pr[:id]), new_pr) }
```

Comment:
This part of the method was kinda confusing for me, would you mind elaborating on what these lines of code are doing?

Reply:
The purpose of this method is:
- Get all valid Contributions
- Fetch their current state from GH API
- Return only Contributions that had no error on fetching and changed the pr_state since last fetching

Reply:
> Fetch their current state from GH API

the fetching of the current state is done through the `new_pr_state` method, correct? if so I'd suggest renaming the method to something that describes this process more accurately (maybe `fetch_pr_state`)

Reply:
Changed!